### PR TITLE
Paneltours

### PR DESCRIPTION
**Pull request by ElySeraidarian**
This PR is related to #24, this not all operationnal yet but it will be easier to discuss about it and make the fixes.

I have been able to add most of the panels asked HOWEVER : 
 - Some panels are not displayed at the right place for some reason ( middle of the screen) -> this does not concerns selectInput items. (This issue is also present in plotRowTree)
- Feels like every single panel of plotRowTree is mapped (when launching the app every question mark has a panel) so I was wondering which one should I be adding
- I have no idea where is the RDA plot so this is pretty hard to know if this is working & having accurate description

### CODE REVIEW

**Comment by ElySeraidarian** [comment]
By the way, macOS build was successul (ubuntu still not working but my side issue I will try working on different branch next time)

**Comment by RiboRings** [comment]
Hi, thanks for the PR!

1. Great catch! It turns out .selectInput is a little bit special. If you check the source of for example [reducedDimPlot panel](https://github.com/iSEE/iSEE/blob/devel/R/panel_ComplexHeatmapPlot.R), you will see that other tours don't need the "selectize" string in the title. I added a working example in the checkbox observer of the AbundancPlot panel, you can do for the rest.

2. You are right! Would you like to try to make a tour sequence? For example, if you click the question mark on the upper right corner of reducedDimPlot, it will guide you through all the parameters sequentially. I haven't figured it out myself yet, but it would be cool to have (no need to do if you don't find it interesting).

3. You can see the example in the [RDAPlot vignettes](https://microbiome.github.io/iSEEtree/reference/RDAPlot.html) to make it appear.

Extra: could you keep lines less than 50 characters (vertical line in RStudio) and use indentation with multiples of 4 spaces? These are Bioconductor standards, you can run `BiocCheck::BiocCheck(".")` to see more tips/good practices.

File: R/class-AbundanceDensityPlot.R [hunk 1]
```diff
@@ -229,13 +229,49 @@ setMethod(".multiSelectionResponsive", "AbundanceDensityPlot",
         return(FALSE)
 })
 
+#' @importFrom methods callNextMethod
+#' @importFrom iSEE .getEncodedName .addTourStep
+setMethod(".definePanelTour", "AbundanceDensityPlot", function(x) {
```
Comment (RiboRings) [inline R/class-AbundanceDensityPlot.R:229]: keep lines shorter than 50 characters and use indentation with multiples of 4 spaces, here and everywhere else

File: R/class-AbundancePlot.R [hunk 2]
```diff
@@ -177,11 +177,40 @@ setMethod(".multiSelectionResponsive", "AbundancePlot",
...
 #' @importFrom SummarizedExperiment rowData
 .create_visual_box_for_abund_plot <- function(x, se) {
   
     panel_name <- .getEncodedName(x)
+    
+    .addSpecificTour(class(x)[1], "rank", function(panel_name) {
+      data.frame(rbind(c(element = paste0("#", panel_name,
+                                          "_rank + .selectize-control"), intro = "Here, we can select the
+            taxonomic rank.")))})
+    .addSpecificTour(class(x)[1], "add_legend", function(panel_name) {
```
Comment (RiboRings) [inline R/class-AbundancePlot.R:207]: Fix tours so they don't appear in the center. This line is a working solution.

**Comment by ElySeraidarian** [comment]
First of all thanks for this useful return !

I have made adjustments to make sure BiocCheck runs with the lowest amount of warnings (line lengths & indentation).

Thanks to your example I have been able to place panels at the right place for almost every plot however I still have one issue : it does not work for some parameters of RDAPlot. I believe this is because we are using fields like "add.ellipse" and not something like "add_ellipse" because it's the same code but works for "colour_by" and not "add.ellipse" so I believe this would be the only possibility. 

Moreover, as said previously I also see a tour sequence for RowTree Plot as in reducedDimPlot panel so maybe I'm really missing something... Sorry for that.

**Comment by RiboRings** [comment]
Great job!

The panel tours for RDAPlot are correctly placed now. Using \ to escape the dot fixed the issue.

I think you are right that all panel tour sequences are already there.

You can do a final check and label the review points as solved. When you are satisfied with everything, feel free to merge to devel.

**Pull request merged**